roles/network/tasks/detected_network.yml

```
@@ -101,6 +101,16 @@
   set_fact:
     num_wifi_interfaces: "{{ count_wifi_interfaces.stdout|int }}"
 
+- name: Check for Access Point capablility with 'iw list'
+  command: iw list | grep -v AP: | grep AP | wc -l
```

Sometimes `AP` appears in comments like the following &mdash; e.g. on a Raspberry Pi 4 Model B Rev 1.1 in this case:

```
root@box:~# iw list | grep -vn AP: | grep AP
21:              * AP
164:             * #{ managed } <= 1, #{ AP } <= 1, #{ P2P-client } <= 1, #{ P2P-device } <= 1,
188:            * [ 4WAY_HANDSHAKE_AP_PSK ]: AP mode PSK offload support
```

Should we... perhaps instead... grep for any string(s) `* AP` that appear at the end of any lines?

(If that indeed this is / these are the decider/clincher?)

```suggestion
  command: iw list | grep '* AP$'
```

---

Lines 164 188 will not exist without line 21 and the test below is against an integer, what return value is provided without wc -l?

---

> Lines 164 188 will not exist without line 21

For sure, I agree.

I was just suggesting a failsafe &mdash; e.g. in case the string 'AP' mysteriously appears in comments arising from diverse/wonky WiFi cards in future?

Towards possibly making things even more resilient?

> what return value is provided without wc -l?

I restored the `wc -l` into the suggestion above a few minutes ago.

bash can do the counting, if counting's required, that works too.

Whereas another option is to avoid bash and have Ansible do the counting &mdash; e.g. using code like `REGISTER_VAR.stdout_lines | length > 0` here:

https://github.com/iiab/iiab/blob/db1531eef57091083dd40f9612c309b7af64019e/roles/1-prep/tasks/hardware.yml#L23-L33